### PR TITLE
virtual/workspaces: do not filter if able to get all

### PR DESCRIPTION
**Pull request by sttts**
Admins or users in their home workspace can see all workspaces. Filtering is not necessary, and until GC is implemented (see https://github.com/kcp-dev/kcp/pull/1774) even wrong.

Fixes https://github.com/kcp-dev/kcp/issues/1723.

### CODE REVIEW

File: pkg/virtual/workspaces/registry/rest.go [hunk 1]
```diff
@@ -370,3 +419,34 @@ func (s *REST) Delete(ctx context.Context, name string, deleteValidation rest.Va
...
+			if ev.Object == nil {
+				w.ch <- ev
+				continue
+			}
+			if cws, ok := ev.Object.(*tenancyv1alpha1.ClusterWorkspace); ok {
+				ws := &tenancyv1beta1.Workspace{}
+				projection.ProjectClusterWorkspaceToWorkspace(cws, ws)
+				ev.Object = ws
+			}
+			w.ch <- ev
```
Comment (stevekuznetsov) [inline pkg/virtual/workspaces/registry/rest.go:442]: I imagine we want to have the same kind of handling here for failing to send within some reasonable amount of time as we do in the watcher itself, noi?

Comment (sttts) [reply pkg/virtual/workspaces/registry/rest.go:442]: which is? What can we do here other than panic?

**Review by stevekuznetsov — APPROVED** [review]

**Comment by sttts** [comment]
Pure rebase.

**Review by stevekuznetsov — APPROVED** [review]

**Pull request merged**